Content.Server/Nuke/NukeCodePaperSystem.cs

```diff
@@ -88,20 +96,28 @@ public bool SendNukeCodes(EntityUid station)
 
             var owningStation = station ?? _station.GetOwningStation(uid);
 
+            var codesMessage = new FormattedMessage();
             // Find the first nuke that matches the passed location.
-            foreach (var nuke in EntityQuery<NukeComponent>())
+            var query = EntityQuery<NukeComponent>().ToList();
+            _random.Shuffle(query);
+            foreach (var nuke in query)
             {
-                if (owningStation == null && nuke.OriginMapGrid != (transform.MapID, transform.GridUid)
-                    || nuke.OriginStation != owningStation)
+                if (!getAllCodes &&
+                    (owningStation == null &&
+                    nuke.OriginMapGrid != (transform.MapID, transform.GridUid) ||
+                    nuke.OriginStation != owningStation))
                 {
                     continue;
                 }
 
-                nukeCode = Loc.GetString("nuke-codes-message", ("name", MetaData(nuke.Owner).EntityName), ("code", nuke.Code));
-                return true;
+                codesMessage.PushNewline();
+                codesMessage.AddMarkup(Loc.GetString("nuke-codes-list", ("name", MetaData(nuke.Owner).EntityName), ("code", nuke.Code)));
+                break;
```

Comment:
this will always end the for loop at the first iteration that passes the continue check above, no matter if getAllCodes is true or not

Reply:
i was told to change it so that it only grabs a single code, either the station one or the nukie's one.
I renamed the variables to clarify what it's meant to do.
it only ever grabs a single code, but the variable decides whether or not if that should include off-station nukes.

Reply:
`getRandomCode`?